src/lib/es5.d.ts

```diff
@@ -1236,6 +1236,14 @@ interface Array<T> {
     map<U>(callbackfn: (this: void, value: T, index: number, array: T[]) => U, thisArg: undefined): U[];
     map<Z, U>(callbackfn: (this: Z, value: T, index: number, array: T[]) => U, thisArg: Z): U[];
     /**
+     * Returns the elements of an array that meet the condition specified in a callback function.
+     * @param callbackfn A function that accepts up to three arguments. The filter method calls the callbackfn function one time for each element in the array.
+     * @param thisArg An object to which the this keyword can refer in the callbackfn function. If thisArg is omitted, undefined is used as the this value.
+     */
+    filter<S extends T>(callbackfn: (this: void, value: T, index: number, array: T[]) => value is S): S[];

```

Comment:
Ad first: ReadonlyArray is already updated (see commit comment), so this commit actually brings both classes in sync.

Ad second: I'm not sure if this is the same, as Z is void in the first case and undefined in second. And tt is used the same way 5 lines below the change.